components/collator/src/provider.rs

```diff
@@ -117,7 +117,7 @@ fn data_ce_to_primary(data_ce: u64, c: char) -> u32 {
 #[icu_provider::data_struct(marker(
     CollationDataV1Marker,
     "collator/data@1",
-    fallback_by = "collation",
+    fallback_by = "language",
```

Comment:
Nit: Add a TODO pointing to #3867 that this should use script fallback where Hant goes to Hani.